guide/concepts/bins.md

```diff
@@ -0,0 +1,35 @@
+# Bins
+
+Bins, similarly to pipelines, are containers for elements. However, at the same time,
+they can be placed and linked within pipelines. Although bin is a separate
+Membrane entity, it can be perceived as a pipeline within an element.
+Bins can be nested within one another.
+
+There are two main reasons why bins are useful:
+- they enable creating reusable element groups,
+- they allow managing elements inside, for instance by dynamically spawning
+or replacing them as the stream changes.
+
+## Pads
+
+Bin's pads are defined similarly to element's pads and can be linked the same way.
+However, their role is limited to proxy the stream to elements inside (inputs)
+or outside (outputs). To achieve that, each input pad of a bin needs to be linked
+to an output outside and another input inside the bin. Accordingly, each bin's output
```

Comment:
As this sentence is quite confusing I would be more explicit here, more or less like this:
```suggestion
to both an output pad from the outside of a bin and an input pad of an element inside the bin. Accordingly, each bin's output
```